### ISSUE TITLE
⌨ Headings label missing applicant dialog

### ISSUE THREAD

**Issue by BrookeCameron** [GCTC-NTGC/gc-digital-talent#13400]
### ⌨ Accessibility and Expected Behaviour

Headings / subheadings in the "Review your application" dialog are not labeled in a heading level (i.e. job title, classification etc). 

Copy is read as "clickable" - but this might be related to the focus issue #13395 

### 📋 Steps to Reproduce

1. with VoiceOver navigate using the arrow keys through the "Review your application" dialog

### What operating systems are you encountering the problem on?

macOS

### What browsers are you encountering the problem on?

Firefox

### What assistive technologies are you encountering the problem on?

VoiceOver

### 📸 Screenshot

https://github.com/user-attachments/assets/d37d3145-df8a-41da-93be-6ddb40d14b70

**Comment by petertgiles** [comment]
I think this is the same issue as #13389.   Feel free to repoen if you disagree.
